From 67c55fcba1e9ff24d0adb288eee34c08d4c86e29 Mon Sep 17 00:00:00 2001
From: Caesar Wang <wxt@rock-chips.com>
Date: Thu, 7 Nov 2019 14:14:13 +0800
Subject: [PATCH] arm64: dts: rockchip: change uart1 for debug port

The SD port is reused by uart2, change the hardware the uart1 for debug.

CL-DEPEND=CL:https://10.10.10.29/c/rk/rkbin/+/88782 rkbin: change uart1 for miniloader [NEW]

Change-Id: Idbc7258798b7ccfcaaf34e88cc6cda5ef4f2ce43
Signed-off-by: Caesar Wang <wxt@rock-chips.com>
---
 arch/arm64/boot/dts/rockchip/rk3399-linux.dtsi       |  6 +++---
 arch/arm64/boot/dts/rockchip/rk3399-opp.dtsi         |  4 ++++
 .../boot/dts/rockchip/rk3399pro-evb-v11-linux.dts    | 12 ++++--------
 3 files changed, 11 insertions(+), 11 deletions(-)

diff --git a/arch/arm64/boot/dts/rockchip/rk3399-linux.dtsi b/arch/arm64/boot/dts/rockchip/rk3399-linux.dtsi
index 45e9575576d7..ab62102a5499 100644
--- a/arch/arm64/boot/dts/rockchip/rk3399-linux.dtsi
+++ b/arch/arm64/boot/dts/rockchip/rk3399-linux.dtsi
@@ -47,7 +47,7 @@
 	compatible = "rockchip,linux", "rockchip,rk3399";
 
 	chosen {
-		bootargs = "earlycon=uart8250,mmio32,0xff1a0000 swiotlb=1 console=ttyFIQ0 rw root=PARTUUID=614e0000-0000 rootfstype=ext4 rootwait coherent_pool=1m";
+		bootargs = "earlycon=uart8250,mmio32,0xff190000 swiotlb=1 console=ttyFIQ0 rw root=PARTUUID=614e0000-0000 rootfstype=ext4 rootwait coherent_pool=1m";
 	};
 
 	reserved-memory {
@@ -77,13 +77,13 @@
 
 	fiq_debugger: fiq-debugger {
 		compatible = "rockchip,fiq-debugger";
-		rockchip,serial-id = <2>;
+		rockchip,serial-id = <1>;
 		rockchip,signal-irq = <182>;
 		rockchip,wake-irq = <0>;
 		rockchip,irq-mode-enable = <1>;  /* If enable uart uses irq instead of fiq */
 		rockchip,baudrate = <1500000>;  /* Only 115200 and 1500000 */
 		pinctrl-names = "default";
-		pinctrl-0 = <&uart2c_xfer>;
+		pinctrl-0 = <&uart1_xfer>;
 	};
 
 	cif_isp0: cif_isp@ff910000 {
diff --git a/arch/arm64/boot/dts/rockchip/rk3399-opp.dtsi b/arch/arm64/boot/dts/rockchip/rk3399-opp.dtsi
index 40de2eab2db4..ee1a29d2375e 100644
--- a/arch/arm64/boot/dts/rockchip/rk3399-opp.dtsi
+++ b/arch/arm64/boot/dts/rockchip/rk3399-opp.dtsi
@@ -348,6 +348,10 @@
 			opp-hz = /bits/ 64 <800000000>;
 			opp-microvolt = <900000>;
 		};
+		opp-933000000 {
+			opp-hz = /bits/ 64 <933000000>;
+			opp-microvolt = <900000>;
+		};
 	};
 };
 
diff --git a/arch/arm64/boot/dts/rockchip/rk3399pro-evb-v11-linux.dts b/arch/arm64/boot/dts/rockchip/rk3399pro-evb-v11-linux.dts
index 2231a144fe79..e0254b6f3509 100644
--- a/arch/arm64/boot/dts/rockchip/rk3399pro-evb-v11-linux.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3399pro-evb-v11-linux.dts
@@ -104,12 +104,12 @@
 
 	fiq_debugger: fiq-debugger {
 		compatible = "rockchip,fiq-debugger";
-		rockchip,serial-id = <2>;
+		rockchip,serial-id = <1>;
 		rockchip,wake-irq = <0>;
 		rockchip,irq-mode-enable = <0>;  /* If enable uart uses irq instead of fiq */
 		rockchip,baudrate = <1500000>;  /* Only 115200 and 1500000 */
 		pinctrl-names = "default";
-		pinctrl-0 = <&uart2c_xfer>;
+		pinctrl-0 = <&uart1_xfer>;
 		interrupts = <GIC_SPI 150 IRQ_TYPE_LEVEL_HIGH 0>;
 	};
 
@@ -307,10 +307,6 @@
 	status = "okay";
 };
 
-&fiq_debugger {
-	pinctrl-0 = <&uart2a_xfer>;
-};
-
 &gmac {
 	phy-supply = <&vcc_phy>;
 	phy-mode = "rgmii";
@@ -970,7 +966,7 @@
 
 &rockchip_suspend {
 	status = "okay";
-	rockchip,sleep-debug-en = <1>;
+	rockchip,sleep-debug-en = <0>;
 	rockchip,sleep-mode-config = <
 		(0
 		| RKPM_SLP_ARMPD
@@ -1003,7 +999,7 @@
 	//sd-uhs-sdr25;
 	//sd-uhs-sdr50;
 	//sd-uhs-sdr104;
-	status = "disabled";
+	status = "okay";
 };
 
 &spi1 {
-- 
2.25.0

